### PR TITLE
Make `hydrate()` hydrate

### PR DESCRIPTION
**Pull request by noencke**
## Description

The `hydrate` test utility does not currently hydrate unhydrated nodes in the initial tree.
This PR updates it so that unhydrated nodes in the initial tree are hydrated just as they are when inserted, or when used in the initial tree of a real SharedTree.

This allows tests to re-use handles to nodes passed into the initial tree of the `hydrate()` function.

### CODE REVIEW

File: packages/dds/tree/src/core/forest/editableForest.ts [hunk 1]
```diff
@@ -51,10 +54,19 @@ export function initializeForest(
 	content: readonly ITreeCursorSynchronous[],
 	revisionTagCodec: RevisionTagCodec,
 	idCompressor: IIdCompressor,
+	visitAnchors = false,
 ): void {
 	assert(forest.isEmpty, 0x747 /* forest must be empty */);
 	const delta: DeltaRoot = deltaForRootInitialization(content);
-	applyDelta(delta, forest, makeDetachedFieldIndex("init", revisionTagCodec, idCompressor));
```
Comment (noencke) [inline packages/dds/tree/src/core/forest/editableForest.ts:57]: @Abe27342 FYI all that `applyDelta` does is 
1. acquire a visitor from the forest
2. call `visitDelta` 
3. free the visitor

So we're still doing the exact same thing in the case that `visitAnchors` is false.

File: packages/dds/tree/src/core/forest/editableForest.ts [hunk 2]
```diff
@@ -40,8 +41,10 @@ export interface IEditableForest extends IForestSubscription {
 }
 
 /**
- * Sets the contents of the forest via delta.
- * Requires the fores starts empty.
+ * Initializes the given forest with the given content.
+ * @remarks This does not perform an edit in the typical sense.
+ * Instead, it creates a delta expressing a creation and insertion of the `content` under the {@link rootFieldKey}, and then applies the delta to the forest.
+ * If `visitAnchors` is enabled, then the delta will also be applied to the forest's {@link AnchorSet} (in which case there must be no existing anchors when this function called).
```
Comment (Abe27342) [inline packages/dds/tree/src/core/forest/editableForest.ts:47]:
```suggestion
 * If `visitAnchors` is enabled, then the delta will also be applied to the forest's {@link AnchorSet} (in which case there must be no existing anchors when this function is called).
```

File: packages/dds/tree/src/core/forest/editableForest.ts [hunk 3]
```diff
@@ -51,10 +54,19 @@ export function initializeForest(
...
 	idCompressor: IIdCompressor,
+	visitAnchors = false,
 ): void {
 	assert(forest.isEmpty, 0x747 /* forest must be empty */);
 	const delta: DeltaRoot = deltaForRootInitialization(content);
-	applyDelta(delta, forest, makeDetachedFieldIndex("init", revisionTagCodec, idCompressor));
+	let visitor = forest.acquireVisitor();
+	if (visitAnchors) {
+		assert(forest.anchors.isEmpty(), "anchor set must be empty");
+		const anchorVisitor = forest.anchors.acquireVisitor();
```
Comment (Abe27342) [inline packages/dds/tree/src/core/forest/editableForest.ts:64]: this needs to be freed, doesn't it?

Comment (noencke) [reply packages/dds/tree/src/core/forest/editableForest.ts:64]: nah

Comment (Abe27342) [reply packages/dds/tree/src/core/forest/editableForest.ts:65]: thanks. missed that :)

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**